apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java

```diff
@@ -767,31 +768,41 @@ public void uploadNode(final WebScriptRequest multiPart, final WebScriptResponse
         final String finalType = type;
         final MetadataChanges finalMetadata = metadata;
         final Boolean finalExtractMetadata = extractMetadata;
-        Object resultRef = transactionHelper
-                .doInTransaction(new RetryingTransactionHelper.RetryingTransactionCallback<Object>() {
-                    @Override
-                    public Object execute() throws Throwable {
-                        NodeRef newNode = nodeService
-                                .createNode(new NodeRef(finalParent), finalFile.getFilename(),
-                                        new QName(finalType));
-                        nodeService.setContent(newNode, finalFile.getInputStream(), finalFile.getFilename());
-
-                        if (finalMetadata != null) {
-                            nodeService.setMetadata(newNode, finalMetadata);
-                        }
+        Object resultRef = null;
+        try {
+            resultRef = transactionHelper
+                    .doInTransaction(new RetryingTransactionHelper.RetryingTransactionCallback<Object>() {
```

Comment:
Maybe `transactionHelper.doInTransaction(() -> {` ? I feel like shaving off some levels of indentation would make it more legible.